runners/core-java/src/main/java/org/apache/beam/runners/core/SplittableParDo.java

```diff
@@ -339,17 +540,17 @@ public PaneInfo pane() {
         }
 
         public void output(OutputT output) {
-          baseContext
-              .windowingInternals()
-              .outputWindowedValue(
-                  output, element.getTimestamp(), element.getWindows(), element.getPane());
+          outputManager.output(
```

Comment:
To remove the dependence on `OutputManager`, here are ideas:

 - Can you make the output type simply be `WindowedValue` (with appropriate compensation afterwards). That would be an accurate description of the `DoFn` anyhow.
 - If you need to output to a particular window, `ReduceFnRunner` also needs a one-method interface with `outputWindowedValue(...)` to remove its dependence on `WindowingInternals` so you could create and use it here.
 - If you need to output a compressed representation, then you'll need a new interface different from the above, as `ReduceFnRunner` only outputs to one window at a time.